src/pages/Signup.js

```diff
@@ -11,17 +11,44 @@ import Typography from "@mui/material/Typography"
 import Container from "@mui/material/Container"
 import { createTheme, ThemeProvider } from "@mui/material/styles"
 import { Link as RouterLink } from "react-router-dom"
+import firebase from "../Firebase/FirebaseConfig"
 
 const theme = createTheme()
 
 export default function SignUp() {
+  const [signup, setSignup] = React.useState({
+    firstName: "",
+    email: "",
+    password: "",
+  })
+  console.log("signup", signup)
+
+  const handleChange = (e) => {
+    setSignup({
+      ...signup,
+      [e.target.name]: e.target.value,
+    })
+  }
+
   const handleSubmit = (event) => {
     event.preventDefault()
-    const data = new FormData(event.currentTarget)
-    console.log({
-      email: data.get("email"),
-      password: data.get("password"),
-    })
+    //authenticates a user signing up
+    firebase
+      .auth()
+      .createUserWithEmailAndPassword(signup.email, signup.password)
+      .then((userCredential) => {
+        // Signed in
+        var user = userCredential.user
+        console.log("user", user)
+        //adds a user to the firebase users collection
+        firebase.firestore().collection("users").doc(user.uid).set({
+          user: user.email,
```

Comment:
It might make sense to also store the user's name since I don't think that gets saved anywhere else.Try to be as specific as possible when naming pieces of data. It is safe to assume that any entry in the collection "users" would be a "user", so I think a better representation of the data would be something like
```
set({
   username: user.email,
   email: user.email,
   firstName: signup: firstName
})
```

Reply:
Thank you! I added  the new structure  to the new commit.